x-pack/plugins/alerting/server/routes/rule/apis/list_types/transforms/transform_rule_types_response/v1.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RegistryAlertTypeWithAuth } from '../../../../../../authorization';
+import { TypesRulesResponseBodyV1 } from '../../../../../../../common/routes/rule/apis/list_types';
+
+export const transformRuleTypesResponse = (
+  ruleTypes: Set<RegistryAlertTypeWithAuth>
```

Comment:
I talked with @JiaweiWu about RegistryAlertTypeWithAuth Ideally it would be moved to be an application type but it's a complete mess. As long as it's not been used anywhere else, we should be fine. Wdyt?